src/csharp/Microsoft.Spark.Worker.UnitTest/DaemonWorkerTests.cs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Network;
+using Razorvine.Pickle;
+using Xunit;
+
+namespace Microsoft.Spark.Worker.UnitTest
+{
+    public class DaemonWorkerTests
+    {
+        [Fact]
+        public void TestsDaemonWorkerTaskRunners()
+        {
+            ISocketWrapper daemonSocket = SocketFactory.CreateSocket();
+            
+            var taskRunnerNumber = 3;
+            var typedVersion = new Version(Versions.V2_4_0);
+            var daemonWorker = new DaemonWorker(typedVersion);
+            
+            Task.Run(() => daemonWorker.Run(daemonSocket));
+
+            for (var i = 0; i < taskRunnerNumber; ++i)
+            {
+                CreateAndVerifyConnection(daemonSocket);
+            }
+            
+            Assert.Equal(taskRunnerNumber, daemonWorker.CurrentNumTaskRunners);
+        }
+
+        private static void CreateAndVerifyConnection(ISocketWrapper daemonSocket)
```

Comment:
This seems to duplicate the TaskRunnerTests.TestTaskRunner() test.  Can we refactor or merge these two tests ?

Reply:
I can proceed by refactoring the `CreateAndVerifyConnection` method into a `ConnectionHelper` helper class. Or proceeding by removing the  `TestsDaemonWorkerTaskRunners` class, although it will remove a bit coverage of the `Run(ISocketWrapper listener)` and the `StartServer` methods in the `DaemonWorker` class, what's your opinion?

Reply:
I think you can refactor `WriteTestData` similar to `PayloadReader.Read` and reuse them in `TestTaskRunner` and here.

Reply:
I've moved the `WriteTestData` in the `PayloadWriter` utility class.

Reply:
thanks!